### ISSUE TITLE
[InstCombine] Missing fold for `fsqrt(select(b, c1, c2)) => select(b, fsqrt(c1), fsqrt(c2))`

### ISSUE THREAD

**Issue by Kmeakin** [llvm/llvm-project#110591]
```c
float src(bool b) { return sqrt(b ? 4.0 : 1.0); }
float tgt(bool b) { return b ? sqrt(4.0) : sqrt(1.0); }
```

https://godbolt.org/z/xnvc5q7nx
https://alive2.llvm.org/ce/z/9tiFV4

**Comment by RKSimon** [comment]
InstCombinerImpl::FoldOpIntoSelect might be able to handle this?

**Comment by nikic** [comment]
Could you please clarify what you're trying to report here? As far as I can tell, this optimization already occurs. Your src and tgt generate the same assembly.

**Comment by Kmeakin** [comment]
I guess it must have been fixed since I posted the issue. Previously (on clang 18.1.0) it was:
```asm
src:
        fmov    d0, #1.00000000
        fmov    d1, #4.00000000
        tst     w0, #0x1
        fcsel   d0, d1, d0, ne
        fsqrt   d0, d0
        fcvt    s0, d0
        ret

tgt:
        fmov    s0, #1.00000000
        fmov    s1, #2.00000000
        tst     w0, #0x1
        fcsel   s0, s1, s0, ne
        ret
```

**Comment by Kmeakin** [comment]
I will close this issue since it seems to be fixed

**Comment by nikic** [comment]
Ah yeah, this optimization was recently extended to all trivially vectorizable functions, so that covers fsqrt as well.

**Comment by RKSimon** [comment]
FTR this appears to have fixed by #112011 + #112183
